nixos/modules/services/matrix/matrix-alertmanager.nix

```diff
@@ -0,0 +1,92 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.matrix-alertmanager;
+  concatenatedRooms = lib.concatStringsSep "|" config.services.matrix-alertmanager.matrixRooms;
+in
+{
+  meta.maintainers = [ lib.maintainers.erethon ];
+
+  options.services.matrix-alertmanager = {
+    enable = lib.mkEnableOption "matrix-alertmanager";
+    package = lib.mkPackageOption pkgs "matrix-alertmanager" { };
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 3000;
+      description = "Port that matrix-alertmanager listens on.";
+    };
+    homeserverUrl = lib.mkOption {
+      type = lib.types.str;
+      description = "URL of the Matrix homeserver to use.";
+      example = "https://matrix.example.com";
+    };
+    matrixUser = lib.mkOption {
+      type = lib.types.str;
+      description = "Matrix user to use for the bot.";
+      example = "@alertmanageruser:example.com";
+    };
+    matrixRooms = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      description = ''
+        Combination of Alertmanager receiver(s) and rooms for the bot to join.
```

Comment:
Looking at this again, I suppose you could go even more structured here. For example:

```nix
matrixRooms = [
  {
    receivers = ["reciever1", "receiver2"];
    roomId = "!roomid@example.com";
  }
]
```

If you did this, you could even assert that roomId is a `strMatching "^!.+"` to enforce that people use a room ID.

I think this would be consistent with other nixos modules options, but as you're the module author, I defer to you on what you think is best.

Reply:
Makes sense, I didn't realize we make modules be so "constrained", but that's good I guess. Thank you for the suggestion, changes applied.

Reply:
I'm honestly not sure how consistent we are about this sort of thing in nixpkgs. This smells reasonable to me, though!